Exception is thrown when setting igxDatePicker's mode with strict template checking

## Description  
When mode="dropdown" is set to the igxDatePicker component - the following error is thrown when running the application:

 error TS2322: Type '"dropdown"' is not assignable to type 'InteractionMode'.
    <igx-date-picker mode="dropdown" class="datepicker">_

 * igniteui-angular version: 11.0.7
 * browser: N/A

## Steps to reproduce  

1. Create a new project
2. Add igxDatePicker
3. Set mode="dropdown"
4. Run the app

## Result  
 error TS2322: Type '"dropdown"' is not assignable to type 'InteractionMode'.
    <igx-date-picker mode="dropdown" class="datepicker">_

## Expected result  
No exception

--- a/projects/igniteui-angular/src/lib/core/enums.ts
+++ b/projects/igniteui-angular/src/lib/core/enums.ts
@@ -1,8 +1,10 @@
+import { mkenum } from './utils';
 /**
  * This enumeration is used to configure whether the date/time picker has an editable input with drop down
  * or is readonly - the date/time is selected only through a dialog.
  */
-export enum InteractionMode {
-    DropDown = 'dropdown',
-    Dialog = 'dialog'
-}
+export const InteractionMode = mkenum({
+    DropDown: 'dropdown',
+    Dialog: 'dialog'
+});
+export type InteractionMode = (typeof InteractionMode)[keyof typeof InteractionMode];

--- a/projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts
+++ b/projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts
@@ -562,7 +562,7 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
      * ```
      */
     @Input()
-    public mode = InteractionMode.Dialog;
+    public mode: InteractionMode = InteractionMode.Dialog;
 
     /**
      * Gets/Sets whether date should spin continuously or stop when min/max is reached.

--- a/projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts
+++ b/projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts
@@ -365,7 +365,7 @@ export class IgxTimePickerComponent implements
      * @memberof IgxTimePickerComponent
      */
     @Input()
-    public mode = InteractionMode.Dialog;
+    public mode: InteractionMode = InteractionMode.Dialog;
 
     /**
      * Determines the container the popup element should be attached to.

